src/CImath/chalf.cpp

```diff
@@ -0,0 +1,233 @@
+#include <../Imath/half.h>
+
+#include "chalf.h"
+
+static_assert (sizeof (IMATH_NAMESPACE::half) == sizeof (imath_half));
+
+extern "C"
+{
+    short convert (int i);
+    float overflow();
+
+    union uif
+    {
+        unsigned int i;
+        float f;
+    };
+
+    static const unsigned short _eLut[1 << 9] =
+#include "../Imath/eLut.h"
+
+        imath_half imath_half_createFromFloat (float f)
+    {
```

Comment:
Can this not use the C++ version (casting on the way out or forcing a type pun)? It seems to me that we want to *wrap* the C++ functionality rather than repeat it. It's going to be much harder to maintain multiple copies of the same code.

Reply:
That's true. I want to try to balance speed with maintainability, so the plan was to reimplement some things where they were trivial. However, if we want to go the type punning route, then I can go for that.